lib/Dialect/FIRRTL/Transforms/ResolveTraces.cpp

```diff
@@ -126,30 +180,57 @@ struct ResolveTracesPass : public ResolveTracesBase<ResolveTracesPass> {
           })
           .Default([&](auto) { targetFieldID = 0; });
     }
+  }
+
+  /// Internal implementation that updates an Annotation to add a "target" field
+  /// based on the current location of the annotation in the circuit.  The value
+  /// of the "target" will be a local target if the Annotation is local and a
+  /// non-local target if the Annotation is non-local.
+  AnnoPathValue updateTargetImpl(Annotation &anno, FModuleLike &module,
+                                 FIRRTLBaseType type, hw::InnerRefAttr name,
+                                 AnnoTarget target) {
+    SmallString<64> newTarget("~");
+    newTarget.append(module->getParentOfType<CircuitOp>().getName());
+    newTarget.append("|");
 
-    anno.setMember("target", StringAttr::get(module->getContext(), newTarget));
+    SmallVector<InstanceOp> instances;
+
+    if (auto nla = anno.getMember<FlatSymbolRefAttr>("circt.nonlocal")) {
+      hw::HierPathOp path = nlaTable->getNLA(nla.getAttr());
+      for (auto part : path.getNamepath().getValue().drop_back()) {
+        auto inst = cast<hw::InnerRefAttr>(part);
+        instances.push_back(dyn_cast<InstanceOp>(
```

Comment:
Ah, need to make it easier to use IRN -- 100% should be easy to ask for resolution of an inner ref.  Anyway, thanks for working through the current situation (including re:other comments).